### PR TITLE
fix: loadProvider now correctly uses the ethers provider, torus now l…

### LINKED ISSUE #182: Torus connector doesn't work
You can test Torus login here: https://demo.snapshot.page Somehow when user login using Torus the request to get accounts or network from the provider aren't resolving. Something must be wrong in the integration. More info here: https://github.com/snapshot-labs/lock/pull/2

### PR DESCRIPTION
**Pull request by samuveth**
Fixes #182 

Changes proposed in this pull request:
- loadProvider to use the ethers provider instead of the the direct wallet provider.
- 
-

### CODE REVIEW

**Comment by bonustrack** [comment]
@samuveth This seem to break the hook when you change network or account on Metamask. Did you tried to do something like ```delete provider.removeAllListeners``` in the lock connector?

**Comment by samuveth** [comment]
Yeah, I just tried a few things like that without success.

**Comment by samuveth** [comment]
@bonustrack I think this fixes the issues with changing the chain and account, not sure but removeAllListerners should also work. From what I see removeAllListerners is a ethers method, so it should probably be used on the provider that ethers generates, like it is now.

File: src/store/modules/web3.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ const actions = {
   },
   loadProvider: async ({ commit, dispatch }) => {
     try {
-      if (auth.provider.removeAllListeners) auth.provider.removeAllListeners();
+      if (auth.web3.removeAllListeners) auth.web3.removeAllListeners();
```
Comment (bonustrack) [inline src/store/modules/web3.ts:64]: The listeners are in `auth.provider` if you use `removeAllListeners()` on `auth.web3` the listeners don't get removed. You can try to connect with Metamask with your Account 1 then switch to account 2 then 3, you will see `accountsChanged` hook is called multiple time while there should be only one call.

Comment (samuveth) [reply src/store/modules/web3.ts:64]: Yes, I noticed that yesterday and was wondering why it was happening. So then I'd suggest we skip that line when `isTorus` is `true`

Comment (bonustrack) [reply src/store/modules/web3.ts]: @samuveth Yes you can do that

File: src/store/modules/web3.ts [hunk 2]
```diff
@@ -74,7 +75,7 @@ const actions = {
         });
         // auth.provider.on('disconnect', async () => {});
       }
-      console.log('Provider', auth.provider);
+      console.log('Provider', auth.web3);
```
Comment (bonustrack) [inline src/store/modules/web3.ts:78]:
```suggestion
      console.log('Provider', auth.provider);
```

**Pull request merged**